ironfist/src/cpp/resource/resourceManager.cpp

```diff
@@ -155,13 +155,7 @@ font* resourceManager::GetFont(char* filename) {
 #ifndef EDITOR
 
 sample* resourceManager::GetSample(const std::string &filename) {
-	//converting to char*
-	char *charname = new char[filename.size() + 1];
-	std::copy(filename.begin(), filename.end(), charname);
-	charname[filename.size()] = '\0';
-	sample* samp = this->GetSample(charname);
-	delete[] charname;
-	return samp;
+	return this->GetSample((char*)filename.c_str());
```

Comment:
Oh....you're way ahead of me. :)

What makes this less safe? Is it that you can't guarantee that GetSample doesn't modify the contents of c_str, or that it can save a reference to the pointer?